### PR TITLE
feat: add libgroup template and mangalib source

### PR DESCRIPTION
**Pull request by vlapsk1y**
Checklist:
- [x] Updated source's version for individual source changes
- [x] Updated all sources' versions for template changes
- [x] Set appropriate `nsfw` value
- [x] Did not change `id` even if a source's name or language were changed
- [x] Tested the modifications by running it on the simulator or a test device 

I know about this pq (#176), but it's not merged into main branch and soon stay deprecated, because mangalib.me in the near future get very major update (test-front.mangalib.me). The new version of mangalib has become unparseable and open at browser as web app.

### CODE REVIEW

File: src/rust/multi.libgroup/template/src/parser.rs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+                title: title,
+                author: String::new(),
+                artist: String::new(),
+                description: String::new(),
+                url: url,
+                categories: Vec::new(),
+                status: status,
+                nsfw: MangaContentRating::Safe,
+                viewer: aidoku::MangaViewer::Rtl
+            })
```
Comment (Skittyblock) [inline src/rust/multi.libgroup/template/src/parser.rs:54]: use `..Default::default` at the end of the struct and exclude the fields you don't populate
```suggestion
            arr_manga.push(Manga {
                id,
                cover,
                title,
                url,
                status,
                nsfw: MangaContentRating::Safe,
                viewer: aidoku::MangaViewer::Rtl,
                ..Default::default
            })
```

**Review by Skittyblock — CHANGES_REQUESTED** [review]
- jsons should use tabs for indentation 
- files should have trailing newlines
- run `cargo +nightly fmt`
- I'm not sure about the "site id" stuff, especially since you're only adding mangalib. the template struct should probably just provide the url and route and stuff, with the actual sources containing the stuff rather than the template hardcoding it.

**Comment by vlapsk1y** [comment]
as for "site id", I wanted to make the task easier for those who want to implement other sites of libgroup. Although them still need to edit template, cuz nsfw sites has own cdn. I really didn't knew about that fact, until checked before writing the comment.

**Comment by Skittyblock** [comment]
yes but the point of the template is to contain all things specific to the template that the individual websites use, but then it's the job of the actual sources to just use the template while providing the source specific things. for example, the individual source should be able to give any base url to the template, and the template shouldn't need to know which exact source it is.

**Comment by vlapsk1y** [comment]
I understood your point, I'll make a commit in the next days. thanks u.

File: src/rust/multi.libgroup/sources/mangalib/res/filters.json
Comment (Skittyblock) [inline src/rust/multi.libgroup/sources/mangalib/res/filters.json:1]: jsons are still using spaces for indentation instead of tabs

**Comment by Skittyblock** [comment]
icon should be 128x128 and make sure to fix the warnings clippy generates when running `cargo +nightly clippy`, but after that it should be good

File: src/rust/multi.libgroup/sources/mangalib/res/filters.json
Comment (Skittyblock) [inline src/rust/multi.libgroup/sources/mangalib/res/filters.json:1]: ids need to be strings

**Review by Skittyblock — APPROVED** [review]

**Pull request merged**